safe/utilities/analysis_handler.py

```diff
@@ -753,3 +813,38 @@ def open_map_in_composer(self, impact_report):
                 space_between_pages * (number_pages - 1))
             self.composer.fitInView(
                 0, 0, paper_width + 1, height + 1, QtCore.Qt.KeepAspectRatio)
+
+    def generate_impact_report(self, impact_function):
+        """
+
+        :param impact_function:
+        :type impact_function: ImpactFunction
+        :return:
+        """
+        # get minimum needs profile
+        minimum_needs = NeedsProfile()
+        minimum_needs.load()
+
+        # create impact report instance
+        report_metadata = ReportMetadata(
+            metadata_dict=standard_impact_report_metadata)
+        impact_report = ImpactReportV4(
+            self.iface,
+            report_metadata,
+            impact_function=impact_function,
+            minimum_needs_profile=minimum_needs)
+
+        # generate report folder
+
+        # no other option for now
+        # TODO: retrieve the information from data store
+        if isinstance(impact_function.datastore.uri, QDir):
+            layer_dir = impact_function.datastore.uri.absolutePath()
+        else:
+            # No other way for now
+            return
+
+        # We will generate it on the fly without storing it after datastore
+        # supports
+        impact_report.output_folder = os.path.join(layer_dir, 'output')
+        impact_report.process_component()
```

Comment:
All this file looks like a copy/paste from the dock ;-)
Can't we do something here to avoid duplicated code ?